pkg/opsgenieIntegrationManager/client.go

```diff
@@ -0,0 +1,106 @@
+package opsgenieIntegrationManager
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"bitbucket.org/atlassianlabs/restclient"
+	"github.com/atlassian/voyager/pkg/util"
+	"github.com/atlassian/voyager/pkg/util/httputil"
+	"github.com/atlassian/voyager/pkg/util/pkiutil"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+const (
+	asapAudience     = "micros-server"
```

Comment:
if we want to optimize for migration to a separate service in the future, we could also make this configurable.

@jokeyrhyme can you confirm the ASAP audience to be used there? agree with Mikhail that it shouldn't be tied to micros-server.

Reply:
I asked @jokeyrhyme on Slack and he replied that we can set whatever we like. I'll set this and comment.

Reply:
Well, this is... not secure. The server must check audience otherwise someone can use any valid ASAP token to talk to it. e.g. I can generate a token myself and use it. /cc @jokeyrhyme

Reply:
it probably has a whitelist of issuers though? otherwise audience is useless.
@ash2k I think you're confusing issuers and audience.

Reply:
No, I'm not =)

Reply:
> I can generate a token myself and use it.

@ash2k well that's not true, if Opsgenie integration manager has an issuer whitelist (does it?), even if doesn't have an audience check. There is still a potential issue of a token generated by Voyager for deployinator (as an example) being stolen and used for request to Opsgenie though, but it's not as critical.

Reply:
Currently opsgenie integration manager ONLY accepts "micros-server" as the ASAP audience. cc @jokeyrhyme

Reply:
@fcobb what about issuers? Can you double check that issuers are whitelisted (i.e. only voyager synchronizer and micros-server are allowed to talk to Opsgenie integration manager)

Reply:
I ran the manual client test (found in this PR) using the **asap-creator secret** and it worked.

Reply:
Sorry, I was previously tuning out from GitHub notifications, because there are multiple teams worth of activity here, but I've adjusted this now so mentions will stand out :)
I've commented on MICROS-6953 about this, after taking a quick look
We don't check `claims.aud` in our code or dependencies anywhere as far as I can see, but we do check `claims.iss` to assert that it matches

Reply:
@fcobb do creator and synchronization share the same issuer then?

Reply:
@nilebox No.
`kubectl -n voyager get secrets asap-voyager-synchronization` gives me `issuer: a3ViZS92b3lhZ2VyLXN5bmNocm9uaXphdGlvbg==`
`kubectl -n voyager get secrets asap-creator -o yaml` gives me `issuer: a3ViZS92b3lhZ2VyLWNyZWF0b3I=`

@jokeyrhyme "but we do check `claims.iss` to assert that it matches"  What does it match exactly?